src/Adapter/Carrier/QueryHandler/GetAvailableCarriersHandler.php

```diff
@@ -163,12 +166,19 @@ private function getProductName(Product $product): string
     {
         if (is_array($product->name)) {
             $languageId = $this->languageContext->getId();
+            $defaultLanguageId = $this->defaultLanguageContext->getId();
 
             if (!isset($product->name[$languageId])) {
                 throw new RuntimeException(sprintf('Product name not found for product ID %d and language ID %d.', $product->id, $languageId));
             }
 
-            return $product->name[$languageId];
+            $productName = $product->name[$languageId];
+
+            if (empty($productName)) {
+                $productName = $product->name[$defaultLanguageId];
+            }
+
+            return $productName;
```

Comment:
Did you have the chance to try this syntax?


```suggestion
            return $product->name[$languageId] ?? $product->name[$defaultLanguageId];
```

Reply:
I revised my copy, more resilient in case of index problems